pkg/asset/installconfig/powervs/session.go

```diff
@@ -67,28 +58,31 @@ type SessionVars struct {
 	PowerVSResourceGroup string
 }
 
-func authenticateAPIKey(sess *bxsession.Session) error {
-	config := sess.Config
-	tokenRefresher, err := authentication.NewIAMAuthRepository(config, &rest.Client{
-		DefaultHeader: gohttp.Header{
-			"User-Agent": []string{http.UserAgent()},
-		},
-	})
+func authenticateAPIKey(apikey string) (string, error) {
+	a, err := core.NewIamAuthenticatorBuilder().SetApiKey(apikey).Build()
 	if err != nil {
-		return err
+		return "", err
 	}
-	return tokenRefresher.AuthenticateAPIKey(config.BluemixAPIKey)
+	token, err := a.GetToken()
+	if err != nil {
+		return "", err
+	}
+	return token, nil
 }
 
-func fetchUserDetails(sess *bxsession.Session) (*User, error) {
-	config := sess.Config
+func FetchUserDetails(apikey string) (*User, error) {
```

Comment:
I think the linter will complain about the lack of comment docs for exported functions:
```suggestion
// FetchUserDetails fetches user details from an api key
func FetchUserDetails(apikey string) (*User, error) {
```